src/main/java/gov/cabinetoffice/gapfindapiadmin/controllers/ApiKeyController.java

```diff
@@ -100,7 +105,26 @@ public ModelAndView displayError() {
 
     @GetMapping("/manage")
     @PreAuthorize("hasAuthority('SUPER_ADMIN')")
-    public ModelAndView displaySuperAdminPage() {
-        return new ModelAndView(SUPER_ADMIN_PAGE);
+    public ModelAndView displaySuperAdminPage(@RequestParam(value = "selectedDepartments", required = false) List<String> selectedDepartment,
+                                 @RequestParam(value = "page", required = false) Optional<Integer> page) {
+        final List<GapApiKey> allApiKeys = apiKeyService.getApiKeysForSelectedFundingOrganisations(selectedDepartment);
+        final List<String> allFundingOrganisations = apiKeyService.getFundingOrgForAllApiKeys();
+        final Long activeKeyCount = apiKeyService.getActiveKeyCount(allApiKeys);
+
+        final int currentPage = page.orElse(1);
```

Comment:
could move most of this logic in an helper class, and invoke the methods straight into .addObject